packages/cli-repl/src/cli-repl.ts

```diff
@@ -140,7 +140,47 @@ class CliRepl {
       }
     });
 
-    const originalEval = util.promisify(this.repl.eval);
+    type REPLEvalArgs = REPLServer extends {
+      eval(...args: infer T);
+    } ? T : never;
+    const replEval = this.repl.eval.bind(this.repl);
+    const originalEval = util.promisify((...args: REPLEvalArgs) => {
```

Comment:
what do you think to put this whole part with Domain as a separate function maybe with some explanatory name? Kinda feel  like the `start()` method of CLIRepl is already a lot to digest.

Reply:
sure, done!